zhudi/gui.py

```diff
@@ -315,16 +315,7 @@ def display_another_result(self, selection):
         if not self.lock:
             model, treeiter = selection.get_selected()
             if treeiter is not None:
-                row = model[treeiter][0]
-                counter = 0
-                if row is not None:
-                    while row[counter] != ".":
-                        counter += 1
-                    figure = int(row[0:counter])
-                    if figure > len(DICTIONARY_TOOLS_OBJECT.index):
-                        self.display_translation(0)
-                    else:
-                        self.display_translation(figure-1)
+                self.display_translation(model[treeiter].path[0])
```

Comment:
Not sure that line is totally equivalent

Reply:
The old way parses "13. blah blah" as "13" then converts to 12. The new way takes the "path", which seems to be an index into the location on the list, and uses that directly. So the path to the 13th element would just be [12].

Reply:
It might not be equivalent if the numbers ever do something that's not just going in order 1. 2. 3. etc.
I also don't really understand the part `if figure > len(DICTIONARY_TOOLS_OBJECT.index):`
This works for me as far as I can tell, even if there are no results to the search.

Reply:
fair enough.

Side note: I tried starting zhudi locally with gtk4, and… well, it's clearly not ready yet.
I actually started a conversion to use poetry a while back that's not yet finished, but if you think that's interesting, I might try to follow that up.